src/neoxp/Extensions/Extensions.cs

```diff
@@ -266,5 +267,48 @@ static bool TryGetNEP6Wallet(string path, string password, ProtocolSettings sett
                 }
             }
         }
+        
+        
+        private static JObject[] MakeFindStatesParams(
+            UInt256 rootHash,
+            UInt160 scriptHash,
+            ReadOnlySpan<byte> prefix,
+            ReadOnlySpan<byte> from = default (ReadOnlySpan<byte>),
+            int? count = null)
+        {
+            JObject[] jobjectArray = new JObject[count.HasValue ? 5 : 4];
+            jobjectArray[0] = (JObject) rootHash.ToString();
+            jobjectArray[1] = (JObject) scriptHash.ToString();
+            jobjectArray[2] = (JObject) Convert.ToBase64String(prefix);
+            jobjectArray[3] = (JObject) Convert.ToBase64String(from);
+            if (count.HasValue)
+                jobjectArray[4] = (JObject) (double) count.Value;
+            return jobjectArray;
+        }
+        public static async Task<ExpressRpcFoundStates> ExpressFindStatesAsync(this RpcClient rpcClient, UInt256 rootHash, UInt160 contractScriptHash, ReadOnlyMemory<byte> prefix, ReadOnlyMemory<byte> from = default, int? pageSize = null)
+        {
+            ExpressRpcFoundStates? states = null;
+            var start = from.ToArray();
```

Comment:
start param should be a `ReadOnlyMemory<byte>` and we can avoid converting the from param to an array.

Reply:
it cannot be readonly because we update `start` on line `310`

Reply:
If we make `start` a `ReadOnlyMemory<byte>` type, we can't modify the contents of `start`, but we can still re-assign the `start` variable to a different `ReadOnlyMemory<byte>` instance (i.e. the return value of `Convert.FromBase64String`). 

Unlike say JavaScript or TypeScript, C# doesn't really have the concept of const or read only local variables. If you specify a const, it has to be a compile time constant, not something you assign at runtime. 

I can see why this is confusing to the non C# dev

Reply:
Mmmh I swear I remember my IDE warning me, but I believe what you're saying. Seems solved now anyway :)